assets/js/components/AnalyticsInactiveCTA.js

```diff
@@ -24,64 +24,60 @@ import PropTypes from 'prop-types';
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useCallback } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import {
-	activateOrDeactivateModule,
-	getReAuthURL,
 	showErrorNotification,
 } from '../util';
-import { refreshAuthentication } from '../util/refresh-authentication';
-import data from './data';
+import Data from 'googlesitekit-data';
 import CTA from './notifications/cta';
 import GenericError from './notifications/generic-error';
+import { STORE_NAME as CORE_USER, PERMISSION_MANAGE_OPTIONS } from '../googlesitekit/datastore/user/constants';
+import { STORE_NAME as CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
 
-class AnalyticsInactiveCTA extends Component {
-	static async setupAnalyticsClick() {
-		try {
-			await activateOrDeactivateModule( data, 'analytics', true );
+const { useSelect, useDispatch } = Data;
 
-			await refreshAuthentication();
+function AnalyticsInactiveCTA( {
+	title = __( 'Learn more about what visitors do on your site.', 'google-site-kit' ),
+	description = __( 'Connect with Google Analytics to see unique visitors, goal completions, top pages and more.', 'google-site-kit' ),
+	ctaLabel = __( 'Set up Analytics', 'google-site-kit' ),
```

Comment:
I think writing out prop destructuring in parameters makes the code unnecessarily hard to read, unless it's for very minimal cases (e.g. 1-3 props without defaults). Generally, let's destructure props in a `const` assignment within a component - this is also an improvement because this way the variables are explicitly `const`.

In this particular case, it can furthermore cause actual problems because translation strings shouldn't be default parameter values - they need to be translated when the function is actually called, otherwise we might still run into the same timing issues @aaemnnosttv was mentioning before.

Reply:
@felixarntz – using the translation strings here as JS default values is safe because these aren't evaluated until the function is called, unlike React `Component.defaultProps` which are evaluated when the module is loaded.
It's essentially the same thing you're suggesting (moving to a destructure with `const { } = props`), just inline.

No problems with that suggestion, but just clarifying that using default values here doesn't have the same risk as when defined outside of the component 👍